File: docs/API/example/Python/MIDI-MT-REMOTE-API.py
import websocket
import _thread
import time
import rel

def on_message(ws, message):
    print(message)

def on_error(ws, error):
    print(error)

def on_close(ws, close_status_code, close_msg):
    print("### closed ###")

def on_open(ws):
    print("Opened connection")
    ws.send("{ \"action\": \"config\" }")

if __name__ == "__main__":
    websocket.enableTrace(True)
    ws = websocket.WebSocketApp("ws://192.168.1.1:8888/data",
                              on_open=on_open,
                              on_message=on_message,
                              on_error=on_error,
                              on_close=on_close)

ws.run_forever(dispatcher=rel, reconnect=5)
rel.signal(2, rel.abort)
rel.dispatch()  

File: docs/API/example/JavaScript/MIDI-MT-REMOTE-API.js

let count = 0;
let printinfo = function (s) {
    let d = document.getElementById("log");
    let b = document.createElement("br");
    count++;
    d.append(b);
    d.append("WS [" + count.toString() + "] " + s);
};

let socket = new WebSocket("ws://192.168.1.1:8888/data");

socket.onopen = function (e) {
    console.log("[open] Connection established, request configuration from server...");
    printinfo("connection open.");
    socket.send(JSON.stringify({ "action": "config" }));
};

socket.onmessage = function (event) {
    console.log("[message]", JSON.parse(event.data));
    printinfo("new message.. see console log.");
};

socket.onclose = function (event) {
    if (event.wasClean) {
        console.log(`[close] Connection closed cleanly, code=${event.code} reason=${event.reason}`);
        printinfo(`connection close: code=${event.code} reason=${event.reason}.`);
    } else {
        console.log("[close] Connection died");
        printinfo("connection died.");
    }
};

socket.onerror = function (error) {
    console.log(`[error]`);
    printinfo(`exception: [error]`);
};
